docs/intro/tutorial07.txt

```diff
@@ -0,0 +1,422 @@
+database. Django knows that a :class:`~django.db.models.ForeignKey` should be
+represented in the admin as a ``<select>`` box. In our case, only one question
+exists at this point.
+
+Also note the "Add Another" link next to "Question." Every object with a
+``ForeignKey`` relationship to another gets this for free. When you click "Add
+Another", you'll get a popup window with the "Add question" form. If you add a question
+in that window and click "Save", Django will save the question to the database and
+dynamically add it as the selected choice on the "Add choice" form you're
+looking at.
+
+But, really, this is an inefficient way of adding ``Choice`` objects to the system.
+It'd be better if you could add a bunch of Choices directly when you create the
+``Question`` object. Let's make that happen.
+
+Remove the ``register()`` call for the ``Choice`` model. Then, edit the ``Question``
+registration code to read:
+
+.. snippet::
+    :filename: polls/admin.py
+
+    from django.contrib import admin
+
+    from .models import Choice, Question
+
+
+    class ChoiceInline(admin.StackedInline):
+        model = Choice
+        extra = 3
+
+
+    class QuestionAdmin(admin.ModelAdmin):
+        fieldsets = [
+            (None,               {'fields': ['question_text']}),
+            ('Date information', {'fields': ['pub_date'], 'classes': ['collapse']}),
+        ]
+        inlines = [ChoiceInline]
+
+    admin.site.register(Question, QuestionAdmin)
+
+This tells Django: "``Choice`` objects are edited on the ``Question`` admin page. By
+default, provide enough fields for 3 choices."
+
+Load the "Add question" page to see how that looks:
+
+.. image:: _images/admin10t.png
+   :alt: Add question page now has choices on it
+
+It works like this: There are three slots for related Choices -- as specified
+by ``extra`` -- and each time you come back to the "Change" page for an
+already-created object, you get another three extra slots.
+
+At the end of the three current slots you will find an "Add another Choice"
+link.  If you click on it, a new slot will be added. If you want to remove the
+added slot, you can click on the X to the top right of the added slot. Note
+that you can't remove the original three slots. This image shows an added slot:
+
+.. image:: _images/admin14t.png
+   :alt: Additional slot added dynamically
+
+One small problem, though. It takes a lot of screen space to display all the
+fields for entering related ``Choice`` objects. For that reason, Django offers a
+tabular way of displaying inline related objects; you just need to change
+the ``ChoiceInline`` declaration to read:
+
+.. snippet::
+    :filename: polls/admin.py
+
+    class ChoiceInline(admin.TabularInline):
+        #...
+
+With that ``TabularInline`` (instead of ``StackedInline``), the
+related objects are displayed in a more compact, table-based format:
+
+.. image:: _images/admin11t.png
```

Comment:
please crop this screenshot to only show "Choices" inlines (Date information is collapsed again)